### PR TITLE
Fix too-many-threads for Python on Travis

### LINKED ISSUE #1037: Python-tests-on-Travis sometimes fail with "error: can't start new thread"
The problem seems probabilistic and reattempting the test is usually enough to get a green result but it's annoying, time-consuming, probably something to do with how Travis constrains the tests run within it, and something that should be fixed at some point.

### PR DESCRIPTION
**Pull request by soltanmm**
Attempting to fix #1037.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -25,7 +25,8 @@ script:
   - rvm use $RUBY_VERSION
   - gem install bundler
   - if [ ! -z "$USE_GCC" ] ; then export CC=gcc-$USE_GCC ; export CXX=g++-$USE_GCC ; fi
-  - ./tools/run_tests/run_tests.py -l $TEST -t -j 16 -c $CONFIG -s 4.0
+  - if [ ! -z "$MAX_JOBS" ] ; then export MAX_JOBS=16 ; fi
```
Comment (murgatroid99) [inline .travis.yml]: This can be done more simply by defining `MAX_JOBS=16` in the global variables at the top.

**Comment by soltanmm** [comment]
Well. Not working: https://travis-ci.org/grpc/grpc/jobs/57105522

_Might_ have to go in and change the size of the created thread pools, which might need some further changes to the Python API.

**Comment by ctiller** [comment]
What's the status with this?

**Comment by ctiller** [comment]
Should this be closed?

**Comment by soltanmm** [comment]
Closing this given the progress we've been making @ctiller's branch.

**Pull request closed without merging**